Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,130 @@
+                workspace.loadRootPackage(at: packageRoot, observabilityScope: tool.observabilityScope, completion: $0)
+            }
+            
+            let possibleCanidates = packageGraph.products.filter { $0.type == .executable }
+            let productToInstall: Product
+            
+            switch possibleCanidates.count {
+            case 0:
+                throw StringError("No Executable Products in Package.swift.")
+            case 1:
+                productToInstall = possibleCanidates[0]
+            default: // More than one, ask the user which one they wanna install
+                print("More than one executable target selected, please select of the following which you'd like to install by typing the number and pressing enter:")
+                for (index, product) in possibleCanidates.enumerated() {
+                    print("[\(index)] \(product.name)")
+                }
+                
+                guard let input = readLine(), let int = Int(input), possibleCanidates.indices.contains(int) else {
+                    throw StringError("Input should be a number between 0 and \(possibleCanidates.count - 1).")
+                }
+                
+                productToInstall = possibleCanidates[int]
+            }
+            
+            if let existingPkg = alreadyExisting.first(where: { $0.name == productToInstall.name }) {
+                throw StringError("\(productToInstall.name) is already installed at \(existingPkg.url)")
+            }
+            
+            try tool.createBuildSystem(explicitProduct: productToInstall.name).build(subset: .product(productToInstall.name))
+            
+            let binPath = try tool.buildParameters().buildPath.appending(component: productToInstall.name)
+            let finalBinPath = swiftpmBinDir.appending(component: binPath.basename)
+            try tool.fileSystem.move(from: binPath, to: finalBinPath)
+            
+            let pkgInstance = InstalledPackage(name: productToInstall.name, url: finalBinPath.asURL)
+            alreadyExisting.append(pkgInstance)
+            try JSONEncoder().encode(alreadyExisting).write(to: registryURL)
+        }
+    }
+    
+    struct Remove: SwiftCommand {
+        
+        @OptionGroup
+        var globalOptions: GlobalOptions
+        
+        @Option(help: "Name of the executable to remove.")
+        var name: String
+        
+        func run(_ tool: SwiftTool) throws {
+            let dotSwiftPMDir = try tool.fileSystem.getOrCreateDotSwiftPMDirectory()
+            let registryURL = dotSwiftPMDir.appending(component: "installedPackages.json").asURL
+            var alreadyRegistered = (try? InstalledPackage.registered(registryURL: registryURL)) ?? []
+            
+            guard let whatWeWantToRemove = alreadyRegistered.first(where: { $0.name == name || $0.url.lastPathComponent == name }) else {
+                throw StringError("No such installed executable as \(name)")
```

Comment:
I'm a bit confused here, is `name` the name of the package or the product? If the former is the case then the string reads wrong.

Reply:
`name`, the variable here, would be specifying either the name of the executable or the product

Reply:
I find this confusing though, `name` property of `InstalledPackage` is not a name of an installed package, but a name of an executable product? We'd have to either rename `InstalledPackage` then and the JSON file that stores it or make it consistent and actually track package names.